diff --git a/vendor/amzn/amazon-pay-module/view/frontend/web/js/view/shipping.js b/vendor/amzn/amazon-pay-module/view/frontend/web/js/view/shipping.js
index c7d8897..6a2c7e0 100644
--- a/vendor/amzn/amazon-pay-module/view/frontend/web/js/view/shipping.js
+++ b/vendor/amzn/amazon-pay-module/view/frontend/web/js/view/shipping.js
@@ -2,22 +2,14 @@
 define(
     [
         'jquery',
-        'underscore',
-        'ko',
         'Magento_Checkout/js/view/shipping',
         'Magento_Customer/js/model/customer',
-        'Magento_Checkout/js/action/set-shipping-information',
-        'Magento_Checkout/js/model/step-navigator',
         'Amazon_Payment/js/model/storage'
     ],
     function (
         $,
-        _,
-        ko,
         Component,
         customer,
-        setShippingInformationAction,
-        stepNavigator,
         amazonStorage
     ) {
         'use strict';
@@ -49,32 +41,20 @@ define(
             },

             /**
-             * New setShipping Action for Amazon Pay to bypass validation
+             * Overridden validateShippingInformation for Amazon Pay to bypass validation
+             *
+             * @inheritDoc
              */
-            setShippingInformation: function () {
-
-                /**
-                 * Set Amazon shipping info
-                 */
-                function setShippingInformationAmazon() {
-                    setShippingInformationAction().done(
-                        function () {
-                            stepNavigator.next();
-                        }
-                    );
+            validateShippingInformation: function () {
+                if (!amazonStorage.isAmazonAccountLoggedIn()) {
+                    return this._super();
                 }

-                if (amazonStorage.isAmazonAccountLoggedIn() && customer.isLoggedIn()) {
-                    setShippingInformationAmazon();
-                } else if (amazonStorage.isAmazonAccountLoggedIn() && !customer.isLoggedIn()) {
-
-                    if (this.validateGuestEmail()) {
-                        setShippingInformationAmazon();
-                    }
-                    //if using guest checkout or guest checkout with amazon pay we need to use the main validation
-                } else if (this.validateShippingInformation()) {
-                    setShippingInformationAmazon();
+                if (!customer.isLoggedIn()) {
+                    return this.validateGuestEmail();
                 }
+
+                return true;
             }
         });
     }