diff -Nuar a/vendor/magento/module-elasticsearch/Model/ResourceModel/Fulltext/Collection/SearchCriteriaResolver.php b/vendor/magento/module-elasticsearch/Model/ResourceModel/Fulltext/Collection/SearchCriteriaResolver.php
--- a/vendor/magento/module-elasticsearch/Model/ResourceModel/Fulltext/Collection/SearchCriteriaResolver.php
+++ b/vendor/magento/module-elasticsearch/Model/ResourceModel/Fulltext/Collection/SearchCriteriaResolver.php
@@ -76,11 +76,13 @@
      */
     public function resolve(): SearchCriteria
     {
-        $this->builder->setPageSize($this->size);
         $searchCriteria = $this->builder->create();
         $searchCriteria->setRequestName($this->searchRequestName);
         $searchCriteria->setSortOrders($this->orders);
         $searchCriteria->setCurrentPage($this->currentPage - 1);
+        if ($this->size) {
+            $searchCriteria->setPageSize($this->size);
+        }
 
         return $searchCriteria;
     }
