diff --git a/vendor/magento/module-webapi-async/Controller/Rest/Asynchronous/InputParamsResolver.php b/vendor/magento/module-webapi-async/Controller/Rest/Asynchronous/InputParamsResolver.php
index 93555559ac9a1..2f46685c6b117 100644
--- a/vendor/magento/module-webapi-async/Controller/Rest/Asynchronous/InputParamsResolver.php
+++ b/vendor/magento/module-webapi-async/Controller/Rest/Asynchronous/InputParamsResolver.php
@@ -70,6 +70,11 @@ class InputParamsResolver
      */
     private $methodsMap;
 
+    /**
+     * @var array
+     */
+    private array $inputData = [];
+
     /**
      * Initialize dependencies.
      *
@@ -156,8 +161,14 @@ public function resolve()
      */
     public function getInputData()
     {
+        if (!empty($this->inputData)) {
+            return $this->inputData;
+        }
+
         if ($this->isBulk === false) {
-            return [$this->inputParamsResolver->getInputData()];
+            $this->inputData = [$this->inputParamsResolver->getInputData()];
+
+            return $this->inputData;
         }
         $inputData = $this->request->getRequestData();
 
@@ -167,7 +178,7 @@ public function getInputData()
             $inputData = array_merge($requestBodyParams, $inputData);
         }
 
-        return array_map(function ($singleEntityParams) {
+        $this->inputData = array_map(function ($singleEntityParams) {
             if (is_array($singleEntityParams)) {
                 $singleEntityParams = $this->filterInputData($singleEntityParams);
                 $singleEntityParams = $this->paramsOverrider->override(
@@ -178,6 +189,8 @@ public function getInputData()
 
             return $singleEntityParams;
         }, $inputData);
+
+        return $this->inputData;
     }
 
     /**
