diff -Nuar a/vendor/magento/module-elasticsearch/Model/ResourceModel/Fulltext/Collection/SearchCriteriaResolver.php b/vendor/magento/module-elasticsearch/Model/ResourceModel/Fulltext/Collection/SearchCriteriaResolver.php
--- a/vendor/magento/module-elasticsearch/Model/ResourceModel/Fulltext/Collection/SearchCriteriaResolver.php
+++ b/vendor/magento/module-elasticsearch/Model/ResourceModel/Fulltext/Collection/SearchCriteriaResolver.php
@@ -80,6 +80,9 @@
         $searchCriteria->setRequestName($this->searchRequestName);
         $searchCriteria->setSortOrders($this->orders);
         $searchCriteria->setCurrentPage($this->currentPage - 1);
+        if ($this->size) {
+            $searchCriteria->setPageSize($this->size);
+        }
 
         return $searchCriteria;
     }
